src/libs/antares/study/parts/short-term-storage/properties.cpp

```diff
@@ -119,6 +121,43 @@ bool Properties::loadKey(const IniFile::Property* p)
     return false;
 }
 
+bool Properties::saveToFolder(const std::string& folder) const
+{
+    const std::string pathIni(folder + SEP + "list.ini");
+
+    // Make sure the folder is created
+    if (!Yuni::IO::Directory::Create(folder))
+    {
+        logs.warning() << "Couldn't create dir for sts: " << folder;
+        return false;
+    }
+
+    logs.debug() << "saving file " << pathIni;
+
+    IniFile ini;
+    IniFile::Section* s = ini.addSection(this->name);
+
+    if (this->injectionNominalCapacity.has_value())
+        s->add("injectionnominalcapacity", this->injectionNominalCapacity.value());
```

Comment:
You could overload `IniFile::Section::add` to take an `std::optional` argument, with this behavior.

https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/1335